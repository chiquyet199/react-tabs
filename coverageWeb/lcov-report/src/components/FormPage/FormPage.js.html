<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/FormPage/FormPage.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/FormPage</a> FormPage.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>30/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.88% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>19/34</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">82.86% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>29/35</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {Component, Fragment} from "react";
import {connect} from "react-redux";
import DynamicForm from "../DynamicForm";
import Button from "../Button";
import {
  PageTitle,
  PageContainer,
  TextParagraph,
  ErrorMessage,
  ScrollContentContainer,
  ScrollButtonBar,
  ScrollScrollView,
} from "../common/layout";
import {SubmissionError} from "redux-form";
import {isEmpty} from "underscore";
import {validate} from "../../utils/validators/common";
import {isWeb} from "../../utils/common";
import {object, bool, string, shape} from "prop-types";
&nbsp;
export class BaseFormPage extends Component {
  state = {
    displayMultiSwitchError: false,
  };
&nbsp;
  withValidation = parentSubmit =&gt; async values =&gt; {
    const {schema, userType = "admin", inviteUser, translations} = this.props;
    const errors = await validate(schema.properties, values);
    <span class="missing-if-branch" title="else path not taken" >E</span>if (!isEmpty(errors)) {
      /*
        `branch_` indicates multiple switches
        check if user selected at least one of them
      */
      <span class="missing-if-branch" title="else path not taken" >E</span>if (Object.keys(errors).some(item =&gt; item.includes("branch_"))) {
        await this.setState({
          displayMultiSwitchError: true,
        });
      }
&nbsp;
      Object.keys(errors).forEach(error =&gt; {
        errors[error] = translations.validationErrors[errors[error]];
      });
&nbsp;
      throw new SubmissionError(errors);
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (schema.customInviteUserValidation &amp;&amp; userType !== "user") {</span>
<span class="cstat-no" title="statement not covered" >      await inviteUser(values);</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    return parentSubmit(values);</span>
  };
&nbsp;
  renderHeaderAndInfo(schema) {
    const headerAndInfo = [];
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (schema.header) {
      headerAndInfo.push(
        &lt;PageTitle
          left="true"
          key={`page_header_${schema.header.toLowerCase().replace(" ", "_")}`}
        &gt;
          {schema.header}
        &lt;/PageTitle&gt;,
      );
    }
    <span class="missing-if-branch" title="else path not taken" >E</span>if (schema.info) {
      headerAndInfo.push(
        &lt;TextParagraph
          left="true"
          key={`page_info_${schema.info.toLowerCase().replace(" ", "_")}`}
        &gt;
          {schema.info}
        &lt;/TextParagraph&gt;,
      );
    }
&nbsp;
    return headerAndInfo;
  }
&nbsp;
  componentDidUpdate(prevProps) {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (prevProps.hasErrors !== this.props.hasErrors &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >this.props.hasErrors)</span> {
<span class="cstat-no" title="statement not covered" >      this.props.submit();</span>
    }
  }
&nbsp;
  render() {
    const {displayMultiSwitchError} = this.state;
    const {
      schema,
      components,
      omitSubmit,
      submit,
      onSubmit,
      submitOnEnter,
      screenSize,
      multiSwitchErrorText,
      extraButtons,
      extraButtonBarButtons,
    } = this.props;
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (schema) {
      const initialValues = this.props.initialValues || schema.initialValues;
      const {
        buttonStyles,
        buttonIcon = "arrow-right",
        buttonText = "Next",
        buttonType = "primary",
        layout,
        marginTop,
      } = schema;
&nbsp;
      let multiSwitchError = null;
&nbsp;
      if (displayMultiSwitchError) {
        multiSwitchError = &lt;ErrorMessage&gt;{multiSwitchErrorText}&lt;/ErrorMessage&gt;;
      }
&nbsp;
      return (
        &lt;ScrollContentContainer size={screenSize}&gt;
          &lt;ScrollScrollView&gt;
            &lt;PageContainer layout={layout} margintop={marginTop}&gt;
              {this.renderHeaderAndInfo(schema)}
            &lt;/PageContainer&gt;
            &lt;PageContainer id="dynamicFormWrapper"&gt;
              &lt;DynamicForm
                initialValues={initialValues}
                schema={schema}
                components={components}
                onSubmit={this.withValidation(onSubmit)}
                submit={submit}
                submitOnEnter={submitOnEnter}
              /&gt;
            &lt;/PageContainer&gt;
          &lt;/ScrollScrollView&gt;
          {multiSwitchError}
          {!omitSubmit &amp;&amp; (
            &lt;Fragment&gt;
              &lt;ScrollButtonBar style={buttonStyles}&gt;
                &lt;Button
                  type={buttonType}
                  right={buttonIcon}
                  middle={buttonText}
                  onClick={submit}
                /&gt;
                {extraButtonBarButtons &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >extraButtonBarButtons}</span>
              &lt;/ScrollButtonBar&gt;
              &lt;ScrollButtonBar style={buttonStyles}&gt;
                {extraButtons &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >extraButtons}</span>
              &lt;/ScrollButtonBar&gt;
            &lt;/Fragment&gt;
          )}
        &lt;/ScrollContentContainer&gt;
      );
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    return null;</span>
  }
}
&nbsp;
export const FormPage = <span class="fstat-no" title="function not covered" >pr</span>ops =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;BaseFormPage {...props}&gt;</span>
    {isWeb ? (
      &lt;form&gt;{props.children}&lt;/form&gt;
    ) : (
      &lt;Fragment&gt;{props.children}&lt;/Fragment&gt;
    )}
  &lt;/BaseFormPage&gt;
);
&nbsp;
FormPage.defaultProps = {
  omitSubmit: false,
};
&nbsp;
FormPage.propTypes = {
  hasErrors: bool,
  schema: object.isRequired,
  omitSubmit: bool,
  userType: string,
  multiSwitchErrorText: string,
  extraButtonBarButtons: object,
  extraButtons: object,
  submitOnEnter: bool,
  translations: shape({
    validationErrors: shape({
      onlyLetters: string.isRequired,
      required: string.isRequired,
      selectAtLeastOne: string.isRequired,
      invalidEmail: string.isRequired,
      emailTaken: string.isRequired,
      onlyNumbers: string.isRequired,
      invalidNumber: string.isRequired,
      invalidPassword: string.isRequired,
      passwordsDontMatch: string.isRequired,
      invalidPhoneNumber: string.isRequired,
    }).isRequired,
  }).isRequired,
};
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  multiSwitchErrorText: state.locale.translations.common.multiSwitchError,
  translations: {
    validationErrors: state.locale.translations.common.validationErrors,
  },
});
&nbsp;
export default connect(
  mapStateToProps,
  null,
)(FormPage);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
