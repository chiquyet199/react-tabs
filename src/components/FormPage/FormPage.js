import React, {Component, Fragment} from "react";
import {connect} from "react-redux";
import DynamicForm from "../DynamicForm";
import Button from "../Button";
import {
  PageTitle,
  PageContainer,
  TextParagraph,
  ErrorMessage,
  ScrollContentContainer,
  ScrollButtonBar,
  ScrollScrollView,
} from "../common/layout";
import {SubmissionError} from "redux-form";
import {isEmpty} from "underscore";
import {validate} from "../../utils/validators/common";
import {isWeb} from "../../utils/common";
import {object, bool, string, shape} from "prop-types";

export class BaseFormPage extends Component {
  state = {
    displayMultiSwitchError: false,
  };

  withValidation = parentSubmit => async values => {
    const {schema, userType = "admin", inviteUser, translations} = this.props;
    const errors = await validate(schema.properties, values);
    if (!isEmpty(errors)) {
      /*
        `branch_` indicates multiple switches
        check if user selected at least one of them
      */
      if (Object.keys(errors).some(item => item.includes("branch_"))) {
        await this.setState({
          displayMultiSwitchError: true,
        });
      }

      Object.keys(errors).forEach(error => {
        errors[error] = translations.validationErrors[errors[error]];
      });

      throw new SubmissionError(errors);
    }

    if (schema.customInviteUserValidation && userType !== "user") {
      await inviteUser(values);
    }

    return parentSubmit(values);
  };

  renderHeaderAndInfo(schema) {
    const headerAndInfo = [];

    if (schema.header) {
      headerAndInfo.push(
        <PageTitle
          left="true"
          key={`page_header_${schema.header.toLowerCase().replace(" ", "_")}`}
        >
          {schema.header}
        </PageTitle>,
      );
    }
    if (schema.info) {
      headerAndInfo.push(
        <TextParagraph
          left="true"
          key={`page_info_${schema.info.toLowerCase().replace(" ", "_")}`}
        >
          {schema.info}
        </TextParagraph>,
      );
    }

    return headerAndInfo;
  }

  componentDidUpdate(prevProps) {
    if (prevProps.hasErrors !== this.props.hasErrors && this.props.hasErrors) {
      this.props.submit();
    }
  }

  render() {
    const {displayMultiSwitchError} = this.state;
    const {
      schema,
      components,
      omitSubmit,
      submit,
      onSubmit,
      submitOnEnter,
      screenSize,
      multiSwitchErrorText,
      extraButtons,
      extraButtonBarButtons,
    } = this.props;

    if (schema) {
      const initialValues = this.props.initialValues || schema.initialValues;
      const {
        buttonStyles,
        buttonIcon = "arrow-right",
        buttonText = "Next",
        buttonType = "primary",
        layout,
        marginTop,
      } = schema;

      let multiSwitchError = null;

      if (displayMultiSwitchError) {
        multiSwitchError = <ErrorMessage>{multiSwitchErrorText}</ErrorMessage>;
      }

      return (
        <ScrollContentContainer size={screenSize}>
          <ScrollScrollView>
            <PageContainer layout={layout} margintop={marginTop}>
              {this.renderHeaderAndInfo(schema)}
            </PageContainer>
            <PageContainer id="dynamicFormWrapper">
              <DynamicForm
                initialValues={initialValues}
                schema={schema}
                components={components}
                onSubmit={this.withValidation(onSubmit)}
                submit={submit}
                submitOnEnter={submitOnEnter}
              />
            </PageContainer>
          </ScrollScrollView>
          {multiSwitchError}
          {!omitSubmit && (
            <Fragment>
              <ScrollButtonBar style={buttonStyles}>
                <Button
                  type={buttonType}
                  right={buttonIcon}
                  middle={buttonText}
                  onClick={submit}
                />
                {extraButtonBarButtons && extraButtonBarButtons}
              </ScrollButtonBar>
              <ScrollButtonBar style={buttonStyles}>
                {extraButtons && extraButtons}
              </ScrollButtonBar>
            </Fragment>
          )}
        </ScrollContentContainer>
      );
    }

    return null;
  }
}

export const FormPage = props => (
  <BaseFormPage {...props}>
    {isWeb ? (
      <form>{props.children}</form>
    ) : (
      <Fragment>{props.children}</Fragment>
    )}
  </BaseFormPage>
);

FormPage.defaultProps = {
  omitSubmit: false,
};

FormPage.propTypes = {
  hasErrors: bool,
  schema: object.isRequired,
  omitSubmit: bool,
  userType: string,
  multiSwitchErrorText: string,
  extraButtonBarButtons: object,
  extraButtons: object,
  submitOnEnter: bool,
  translations: shape({
    validationErrors: shape({
      onlyLetters: string.isRequired,
      required: string.isRequired,
      selectAtLeastOne: string.isRequired,
      invalidEmail: string.isRequired,
      emailTaken: string.isRequired,
      onlyNumbers: string.isRequired,
      invalidNumber: string.isRequired,
      invalidPassword: string.isRequired,
      passwordsDontMatch: string.isRequired,
      invalidPhoneNumber: string.isRequired,
    }).isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  multiSwitchErrorText: state.locale.translations.common.multiSwitchError,
  translations: {
    validationErrors: state.locale.translations.common.validationErrors,
  },
});

export default connect(
  mapStateToProps,
  null,
)(FormPage);
