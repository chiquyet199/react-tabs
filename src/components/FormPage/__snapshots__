/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/FormPage/__native_snapshots__