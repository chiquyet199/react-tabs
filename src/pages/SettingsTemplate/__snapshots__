/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/pages/SettingsTemplate/__native_snapshots__