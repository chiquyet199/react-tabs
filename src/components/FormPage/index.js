import {connect} from "react-redux";
import {submit} from "redux-form";
import FormPage from "./FormPage";
import {go} from "../../actions/navigation";

export const mapStateToProps = state => ({
  screenSize: state.dimensions.size,
});

const mapDispatchToProps = {
  submit: () => dispatch => dispatch(submit("DynamicForm")),
  go,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(FormPage);
