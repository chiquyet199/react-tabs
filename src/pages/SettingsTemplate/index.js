import React from "react";
import {func, object} from "prop-types";
import FormPage from "../../components/FormPage";

export const SettingsTemplate = ({schema, initialValues, onSubmit}) => {
  return (
    <FormPage
      schema={schema}
      onSubmit={onSubmit}
      initialValues={initialValues}
    />
  );
};

SettingsTemplate.propTypes = {
  schema: object.isRequired,
  initialValues: object,
  onSubmit: func.isRequired,
};

export default SettingsTemplate;
