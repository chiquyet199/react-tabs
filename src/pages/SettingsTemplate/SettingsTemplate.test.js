import React from "react";
import {shallow} from "enzyme";
import {extend} from "underscore";
import SettingsTemplate from "./index.js";
import FormPage from "../../components/FormPage";

describe("About You Agency Settings - page tests", () => {
  it("matches snapshot", () => {
    expect(
      getComponent({
        schema: getMockSchema(),
      }),
    ).toMatchSnapshot();
  });

  it("renders without initial values", () => {
    expect(
      getComponent({
        schema: getMockSchema(),
        initialValues: null,
      }),
    ).toMatchSnapshot();
  });

  it("Submits Form with initial values", () => {
    const onSubmit = jest.fn();
    const props = getMockProps();
    const comp = getComponent({
      schema: getMockSchema(),
      onSubmit,
    });
    comp.find(FormPage).prop("onSubmit")(props.initialValues);
    comp.update();
    expect(onSubmit).toHaveBeenCalledWith(props.initialValues);
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<SettingsTemplate {...parsedProps} />);
  };

  const getMockProps = () => ({
    initialValues: {name: "Test name"},
    match: {url: "test"},
    updateAgency: () => {},
    onSubmit: () => {},
  });

  const getMockSchema = () => ({
    type: "object",
    layout: "hq",
    header: "Test About header",
    info: "Test About info",
    buttonText: "Save",
    buttonIcon: null,
    buttonStyles: [{width: "165px", alignSelf: "flex-end"}],
    properties: {
      name: {
        type: "string",
        title: "Test name",
      },
    },
  });
});
