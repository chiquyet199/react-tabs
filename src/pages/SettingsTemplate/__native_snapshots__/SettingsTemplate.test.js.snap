// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About You Agency Settings - page tests matches snapshot 1`] = `
<Connect(Connect(FormPage))
  initialValues={
    Object {
      "name": "Test name",
    }
  }
  onSubmit={[Function]}
  schema={
    Object {
      "buttonIcon": null,
      "buttonStyles": Array [
        Object {
          "alignSelf": "flex-end",
          "width": "165px",
        },
      ],
      "buttonText": "Save",
      "header": "Test About header",
      "info": "Test About info",
      "layout": "hq",
      "properties": Object {
        "name": Object {
          "title": "Test name",
          "type": "string",
        },
      },
      "type": "object",
    }
  }
/>
`;

exports[`About You Agency Settings - page tests renders without initial values 1`] = `
<Connect(Connect(FormPage))
  initialValues={null}
  onSubmit={[Function]}
  schema={
    Object {
      "buttonIcon": null,
      "buttonStyles": Array [
        Object {
          "alignSelf": "flex-end",
          "width": "165px",
        },
      ],
      "buttonText": "Save",
      "header": "Test About header",
      "info": "Test About info",
      "layout": "hq",
      "properties": Object {
        "name": Object {
          "title": "Test name",
          "type": "string",
        },
      },
      "type": "object",
    }
  }
/>
`;
